pkgs/tools/misc/rkvm/default.nix

```diff
@@ -0,0 +1,47 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, libevdev
+, openssl
+, makeWrapper
+, nixosTests
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "rkvm";
+  version = "0.4.1";
+
+  src = fetchFromGitHub {
+    owner = "htrefil";
+    repo = pname;
+    rev = version;
+    hash = "sha256-NJ9wvG3UN80OMKQqrOwcdhsypNdJlQB3SaI/wP0DcHM=";
+  };
+
+  cargoHash = "sha256-HvHGOcGKHCuyhd2S4BPketGwo3/Av1WRgbAqBHdP9kw=";
+
+  nativeBuildInputs = [ pkg-config rustPlatform.bindgenHook makeWrapper ];
+  buildInputs = [ libevdev ];
+
+  postInstall = ''
+    install -Dm444 -t "$out/lib/systemd/system" systemd/rkvm-*.service
+    install -Dm444 example/server.toml "$out/etc/rkvm/server.example.toml"
+    install -Dm444 example/client.toml "$out/etc/rkvm/client.example.toml"
+
+    wrapProgram $out/bin/rkvm-certificate-gen --prefix PATH : ${lib.makeBinPath [ openssl ]}
+  '';
+
+  passthru.tests = {
+    inherit (nixosTests) rkvm;
+  };
+
+  meta = with lib; {
+    description = "Virtual KVM switch for Linux machines";
+    homepage = "https://github.com/htrefil/rkvm";
+    changelog = "https://github.com/htrefil/rkvm/releases/tag/${version}";
+    license = licenses.mit;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ ckie ];
+  };
```

Comment:
This should have a `mainProgram` attribute

Reply:
I do not think so.  It has three binaries, and none of them should be the main one.

Reply:
Hmm. Not quite sure. There are lots of other multi-binary packages where there isn't a definitive "main" binary. I.e. `redis` chooses `redis-client` as its main binary.

The README of rkvm mentions this:

> - Since rkvm-server grabs all input, i's a good idea to do a test run first to make sure you won't end up
>   being unable to user your keyboard and/or mouse because your display server is not properly configured to receive input from rkvm.
> 
>   Run the following command to start rkvm-server for 15 seconds to test that your keyboard, mouse, etc. works properly:
>   ```
>   # rkvm-server /etc/rkvm/server.toml --shutdown-after 15
>   ```

So perhaps the server should be the main binary?

Reply:
What about adding mainProgram when a real user case occurs?

Since the author is [busy](https://github.com/NixOS/nixpkgs/pull/250792#issuecomment-1747351363) and this PR is in a good state, I suggest we merge it as is for now.